### ISSUE TITLE
 Call to a member function newSchemaList() on null 

### ISSUE THREAD

**Issue by hexmode** [SemanticMediaWiki/SemanticMediaWiki#5196]
### Setup

- SMW version: master

### Issue

```
Sub Semantic Data (SMW\Tests\DataModel\SubSemanticData)
 ✔ Can construct [0.03 ms]
 ✔ Add sub semantic data [0.18 ms]
 ✔ Add sub semantic data with mismatched subject throws exception [0.68 ms]
 ✔ Remove sub semantic data [0.05 ms]
 ✘ Remove property [0.43 ms]
   │
   │ Error: Call to a member function newSchemaList() on null
   │
   │ /var/www/html/extensions/SemanticMediaWiki/src/DataModel/SequenceMap.php:50
   │ /var/www/html/extensions/SemanticMediaWiki/includes/SemanticData.php:495
   │ /var/www/html/extensions/SemanticMediaWiki/tests/phpunit/DataModel/SubSemanticDataTest.php:116
   │
```

See [test run](https://github.com/SemanticMediaWiki/SemanticMediaWiki/runs/4750905875#step:10:155)

Seems to be related to the order of the tests run.

In this case, [newSchemaFinder](https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/44cc7f39bda4b01f378cb436295f651a2699cdd5/src/Schema/SchemaFactory.php#L156) seems to return null.  WTF?

**Comment by JeroenDeDauw** [comment]
> In this case, newSchemaFinder seems to return null. WTF?

looks impossible

**Comment by hexmode** [comment]
> looks impossible

Agreed. But [it is happening](https://github.com/SemanticMediaWiki/SemanticMediaWiki/runs/4750905875#step:10:155) somehow.

Interestingly, changing the [typehint for newSchemaFinder](https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/5195/commits/1760dfd3175ee6a0cd4cf76fa5c6a66e2fb786ff) moves the failure [down a few lines](https://github.com/SemanticMediaWiki/SemanticMediaWiki/runs/4753611561#step:10:155) because now [newSchemaList](https://github.com/hexmode/SemanticMediaWiki/blob/1760dfd3175ee6a0cd4cf76fa5c6a66e2fb786ff/src/Schema/SchemaFinder.php#L123) is providing null.

The signature of newSchemaList is [@return SchemaList|[]](https://github.com/hexmode/SemanticMediaWiki/blob/1760dfd3175ee6a0cd4cf76fa5c6a66e2fb786ff/src/Schema/SchemaFinder.php#L121) but, if we change it to [@return ?SchemaList](https://github.com/hexmode/SemanticMediaWiki/commit/e9f193fe161da6194de94606aea70238b570e58b), then [the test passes](https://github.com/SemanticMediaWiki/SemanticMediaWiki/runs/4753746024#step:10:155).

I suspect this is a bug in PHP.  It might be fixed in newer PHP

(Note that this test was not run before during CI, which is why we didn't see the failure.)
